### DISCUSSION TITLE
How to disable openssl legacy provider

### DISCUSSION THREAD

**Discussion by old-ratboy** [pocoproject/poco#4058] [Q&A]
Is there a way to disable the check within OpenSSLInitializer::initialize() that tries to load the legacy provider?
`		if (!_legacyProvider)
		{
			_legacyProvider = OSSL_PROVIDER_load(NULL, "legacy");
			if (!_legacyProvider) throw CryptoException("Failed to load OpenSSL legacy provider");
		}
`
The project I am working on doesn't have the legacy provider assemblies for openssl 3.1.X, and I am trying to find the best way to work around this.

**Comment by YJJS321** [comment]
You should set the openssl environment variables.Such as  OPENSSL_HOME,OPENSSL_CONF, OPENSSL_MODULES,and add them to system variables "Path".
